ugvc/vcfbed/vcftools.py

```diff
@@ -561,3 +563,47 @@ def genotype_ordering(num_alt: int) -> np.ndarray:
                 i += 1
     return gr_ar
 
+def replace_data_in_specific_chromosomes(input_vcf: str, new_data_json: str, header_file: str, output_vcf:str, tempdir: str|None = None):
+    """
+    Takes a main vcf file, and a list of modified vcf files, broken down by chromosome (listed in the json).
+    Replaces the data in the main vcf file by the data in the modified vcfs, and returns a concatenated vcf.
+    
+    Parameters
+    ----------
+    input_vcf: The main vcf which is being modified
+    new_data_json: A json file in the form {chromosome: vcf}, with the data that will be replaced in input_json
+    header_file: A file with a valid vcf header that will replace the current header. Make sure the new header fits
+                 the new data.
+    output_vcf: Name of the output vcf
+    tempdir: Name of temp dir
+    """
+    if tempdir is None:
+        tempdir = gettempdir()
+    # Get the modified vcfs
+    with open(new_data_json) as jfile:
+        chrom2vcf = json.load(jfile)
+    # Get info about the contigs
+    contigs_info = {}
+    with pysam.VariantFile(input_vcf) as in_vcf_obj:
+        for k in in_vcf_obj.header.contigs.keys():
+            contig = in_vcf_obj.header.contigs[k]
+            contigs_info[contig.name] = {"id": contig.id, "length": contig.length}
+    sorted_contigs = sorted(contigs_info.keys(), key=lambda k: contigs_info[k]["id"])
+    # bed file with the chromosomes that were not modified by imputation
+    untouched_contigs_bed = os.path.join(tempdir, "untouched_contigs.bed")
+    with open(untouched_contigs_bed, "w", encoding="utf-8") as regions_file:
+        sorted_contigs = sorted(contigs_info.keys(), key=lambda k: contigs_info[k]["id"])
+        for contig in sorted_contigs:
+            if contig not in chrom2vcf.keys():
+                regions_file.write(contig + "\t1\t" + str(contigs_info[contig]["length"]) + "\n")
+    # Extract chromosome without imputation data, and reheader to fit the header of vcfs after imputation
```

Comment:
I think the issue (maybe) is that the imputation changes the header (it should not). Maybe this is the part that we should fix?

Reply:
I add more fields to the vcf, such as the PLs and GTs before the imputation (PL0, GT0 and GQ0) and the DS values that Beagle outputs. So there is a change of the header.